apps/remote_control/lib/lexical/remote_control/project_node.ex

```diff
@@ -0,0 +1,188 @@
+defmodule Lexical.RemoteControl.ProjectNode do
+  alias Lexical.Project
+  alias Lexical.RemoteControl
+  require Logger
+
+  defmodule State do
+    defstruct [
+      :project,
+      :node,
+      :paths,
+      :cookie,
+      :stopped_by,
+      :stop_timeout,
+      :started_by,
+      :status
+    ]
+
+    def new(%Project{} = project) do
+      cookie = Node.get_cookie()
+
+      %__MODULE__{
+        project: project,
+        node: node_name(project),
+        paths: RemoteControl.glob_paths(),
+        cookie: cookie
+      }
+    end
+
+    def set_stopped_by(state, from, stop_timeout) do
+      %{state | stopped_by: from, stop_timeout: stop_timeout}
+    end
+
+    def set_started_by(state, from) do
+      %{state | started_by: from}
+    end
+
+    def set_started(state) do
+      %{state | status: :started}
+    end
+
+    def node_name(%Project{} = project) do
+      :"#{Project.name(project)}@127.0.0.1"
+    end
+  end
+
+  alias Lexical.RemoteControl.ProjectNodeSupervisor
+  use GenServer
+
+  def start(project, project_listener) do
+    node = State.node_name(project)
+    remote_control_config = Application.get_all_env(:remote_control)
+
+    {:ok, node_pid} = ProjectNodeSupervisor.start_project_node(project)
+
+    with :ok <- start_node(project),
+         :ok <-
+           :rpc.call(node, RemoteControl.Bootstrap, :init, [
+             project,
+             project_listener,
+             remote_control_config
+           ]) do
+      {:ok, node_pid}
+    end
+  end
+
+  @start_timeout 5_000
+
+  defp start_node(project) do
+    project |> name() |> GenServer.call(:start, @start_timeout + 500)
+  end
+
+  @stop_timeout 1_000
+
+  def stop(project, stop_timeout \\ @stop_timeout) do
+    project |> name() |> GenServer.call({:stop, stop_timeout}, stop_timeout + 100)
+  end
+
+  def child_spec(project) do
+    %{
+      id: name(project),
+      start: {__MODULE__, :start_link, [project]},
+      restart: :transient
+    }
+  end
+
+  def start_link(project) do
+    state = State.new(project)
+    GenServer.start_link(__MODULE__, state, name: name(project))
+  end
+
+  @impl GenServer
+  def init(state) do
+    {:ok, state}
+  end
+
+  @impl true
+  def handle_call(:start, from, state) do
```

Comment:
please add a `%State{} = ` pattern match on the genserver callbacks to be clear what state is.